CRM/Contribute/Page/ContributionRecur.php

```diff
@@ -61,8 +61,20 @@ public function view() {
       if (!empty($values['contribution_status_id'])) {
         $values['contribution_status'] = CRM_Core_OptionGroup::getLabel('contribution_status', $values['contribution_status_id']);
       }
+      // Get financial type name
+      if (!empty($values['financial_type_id'])) {
+        $values['financial_type_name'] = CRM_Contribute_PseudoConstant::financialType($values['financial_type_id']);
+      }
+      // Get Paid By label
+      if (!empty($values['payment_instrument_id'])) {
+        $values['payment_instrument'] = CRM_Core_OptionGroup::getLabel('payment_instrument', $values['payment_instrument_id']);
```

Comment:
Note this function is now deprecated as it does not use caching. This is the replacement function that is cached
https://github.com/civicrm/civicrm-core/pull/7066/files#diff-9494a1cbe55eb0a702095a6bbcf8ffedL62